### PR TITLE
Clearer handling of ConnectionHandles

### PR DESCRIPTION
**Pull request by djc**
I opted for using `ch` everywhere except as a field on the `ConnectionInner` type in the quinn crate: since that represents a `Connection`, I felt `handle` made the code easier to read.

Also made it so the type is a little more abstracted.

### CODE REVIEW

**Comment by djc** [comment]
Sorry, this is a pretty big rebase hazard. Still makes it easier to read though, IMO!

**Comment by djc** [comment]
(Veering into territory of further decoupling of `Endpoint` and `Connection`.)

File: quinn-proto/src/connection.rs [hunk 1]
```diff
@@ -978,6 +982,40 @@ impl Connection {
...
+        remote: SocketAddr,
+        ecn: Option<EcnCodepoint>,
+        mut remaining: Option<BytesMut>,
+    ) {
+        let loc_cid_len = self.loc_cids.iter().next().unwrap().1.len();
+        while let Some(data) = remaining {
+            match PartialDecode::new(data, loc_cid_len) {
+                Ok((partial_decode, rest)) => {
+                    remaining = rest;
+                    self.handle_decode(now, remote, ecn, partial_decode);
```
Comment (Ralith) [inline quinn-proto/src/connection.rs]: > Receivers SHOULD ignore any subsequent packets with a different Destination Connection ID than the first packet in the datagram.

Something for future work here, I think.

File: quinn-proto/src/endpoint.rs [hunk 2]
```diff
@@ -1031,13 +1031,7 @@ impl slog::Value for Timer {
...
 
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
-pub struct ConnectionHandle(pub usize);
-
-impl From<ConnectionHandle> for usize {
-    fn from(x: ConnectionHandle) -> usize {
-        x.0
-    }
-}
+pub struct ConnectionHandle(usize);
```
Comment (Ralith) [inline quinn-proto/src/endpoint.rs]: I'm making pretty heavy use of this in #198 to skip an unnecessary `HashMap` at the higher level, so I'll probably want to restore it there.

File: quinn-proto/src/connection.rs [hunk 3]
```diff
@@ -978,6 +982,40 @@ impl Connection {
...
+    }
+
+    fn handle_coalesced(
+        &mut self,
+        now: u64,
+        remote: SocketAddr,
+        ecn: Option<EcnCodepoint>,
+        mut remaining: Option<BytesMut>,
+    ) {
+        let loc_cid_len = self.loc_cids.iter().next().unwrap().1.len();
```
Comment (Ralith) [inline quinn-proto/src/connection.rs]: Couldn't this just be `self.config.local_cid_len`, as it was originally?

**Pull request merged**